### PR TITLE
Add the gen-bounds command.

### PR DESCRIPTION
**Pull request by 23Skidoo**
An updated version of #2774 with some mindless refactorings to make it work against current `master`.

### CODE REVIEW

**Comment by mightybyte** [comment]
Cool, thanks!

**Comment by 23Skidoo** [comment]
@mightybyte Sorry for the delay, I still want to get this into the 1.24 or 1.24.1 release of `cabal-install`.

**Comment by mightybyte** [comment]
Great, looking forward to it.

**Comment by ezyang** [comment]
This looks ok to me.

File: cabal-install/Distribution/Client/Freeze.hs [hunk 1]
```diff
@@ -110,11 +100,38 @@ freeze verbosity packageDBs repoCtxt comp platform conf mSandboxPkgInfo
                      "The following packages would be frozen:"
                    : formatPkgs pkgs
 
-             else freezePackages globalFlags verbosity pkgs
+             else freezePackages verbosity globalFlags pkgs
 
   where
     dryRun = fromFlag (freezeDryRun freezeFlags)
 
+getFreezePkgs :: Verbosity
```
Comment (23Skidoo) [inline cabal-install/Distribution/Client/Freeze.hs]: Needs a Haddock comment.

**Comment by 23Skidoo** [comment]
@mightybyte I've added documentation, Haddock comments, and did some cosmetic tweaks to the code. If you're okay with my changes, I'll merge this PR.

**Comment by mightybyte** [comment]
All fine by me.  I'm looking forward to this feature!

**Comment by 23Skidoo** [comment]
Just waiting for Travis to go green now...

**Pull request merged**